lib/tasks/ee.rake

```diff
@@ -0,0 +1,263 @@
+require 'yaml'
+require 'erb'
+
+class Database
+  RECOGNIZED_OPTIONS = %w(user host port socket)
+
+  def initialize(options = {})
```

Comment:
You could still use the keyword args with the params from the yaml file. With `initialize(database:, **options)` it'll raise an error if `database` isn't present, otherwise will put all other params into `options`.

Up to you.